Sources/GeospatialSwift/API/Calculator/GeodesicCalculator.swift

```diff
@@ -334,60 +341,376 @@ extension GeodesicCalculator {
                 guard currentLineIndex < nextLineIndex else { return false }
                 
                 // If next line continues from previous ensure no overlapping
-                if currentLineIndex == nextLineIndex - 1 && currentLineSegment.otherPoint == nextLineSegment.point { return contains(nextLineSegment.otherPoint, in: currentLineSegment, tolerance: tolerance) }
+                if currentLineIndex == nextLineIndex - 1 && currentLineSegment.endPoint == nextLineSegment.startPoint { return contains(nextLineSegment.endPoint, in: currentLineSegment, tolerance: tolerance) }
                 
                 return hasIntersection(currentLineSegment, with: nextLineSegment, tolerance: tolerance)
             }
         }
     }
     
     public func hasIntersection(_ polygon: GeodesicPolygon, tolerance: Double) -> Bool { polygon.linearRings.contains { hasIntersection($0, tolerance: tolerance) } }
+}
+
+// MARK: Violation Indicies
+
+extension GeodesicCalculator {
+    public func simpleViolationDuplicateIndices(points: [GeodesicPoint], tolerance: Double) -> [[Int]] {
+        var uniquePoints = [GeodesicPoint]()
+        var duplicateIndices = [[Int]]()
+        
+        points.enumerated().forEach { pointIndex, point in
+            var isUnique = true
+            for index in (0..<uniquePoints.count) {
+                if distance(from: point, to: uniquePoints[index], tolerance: tolerance) == 0 {
+                    duplicateIndices[index].append(pointIndex)
+                    isUnique = false
+                    break
+                }
+            }
+            if isUnique {
+                uniquePoints.append(point)
+                duplicateIndices.append([pointIndex])
+            }
+        }
+        
```

Comment:
[1,2,3,4]
[5,6,7]
[8]
[9]
[10]
In this case, 8,9,10 are unique. 1 is duplicate with [2,3,4]